overview/content-types/shell-instructions.nix

```
@@ -36,14 +36,21 @@ in
         self:
         if lib.isList self.instructions then
           ''
-            <dl>
-            ${lib.concatMapStringsSep "\n" (i: ''
-              <dt>${i.platform}</dt>
-              <dd>
-                ${toString i.shell-session}
-              </dd>
-            '') self.instructions}
-            </dl>
+            <div class=tabs>
+              {% set input_group_name = unique_id() %}
+              ${lib.concatStringsSep "\n" (
+                lib.zipListsWith (i: instruction: ''
+                  {% set input_id = unique_id() %}
+                  <input id="{{ input_id }}" type="radio" name="{{ input_group_name }}" ${
+                    lib.optionalString (i == 1) "checked"
+                  }>
+                  <label for="{{ input_id }}" >${instruction.platform}</label>
+                  <div class="tab-content">
+                    ${toString instruction.shell-session}
+                  </div>
+                '') (lib.range 1 (lib.length self.instructions)) self.instructions
+              )}
+            </div>
           ''
```

This is a bit more readable for me. What do you think?

```suggestion
          let
            instruction-ids = lib.range 1 (lib.length self.instructions);

            /**
              Return the ith tab input for the ith instruction.
            */
            make-tab-input = i: instruction: ''
              {% set input_id = unique_id() %}
              <input id="{{ input_id }}" type="radio" name="{{ input_group_name }}" ${
                lib.optionalString (i == 1) "checked"
              }>
              <label for="{{ input_id }}" >${instruction.platform}</label>
              <div class="tab-content">
              ${toString instruction.shell-session}
              </div>
            '';
          in
          ''
            <div class=tabs>
              {% set input_group_name = unique_id() %}
              ${lib.concatStringsSep "\n" (lib.zipListsWith make-tab-input instruction-ids self.instructions)}
            </div>
          ''
```

---

Honestly, no.

---

> `i` traditionally stands for index not id. Now it's confused with the IDs for HTML.

Well, it does correspond to the tab `id`, so how about renaming it to that?

> The separate variable makes it seem like the "ids" are all important when they aren't.

To me, it's rather nice describing what it is. When I looked at it the first time, I had to kinda decrypt what it's trying to do: take the length of the instructions and generate a range from 1 to that length, which still doesn't tell you what that range represents.

This said, this isn't a blocker for me.

> input_group_name isn't defined where it's used.

That's a fair point.

> The comment seems superfluous/giving the wrong detail.

> make-tab-input isn't an accurate name. Maybe just make-tab.

Yes, naming things is hard so I just put whatever came to mind ;)

> The most confusing part to me previously was using ziplistswith to get indexes, but now it feels even more unclear why its there.

So, instead of changing the code, how about we add a comment that describes what's happening?

Something in the lines of:
- we generate a unique id for each table
- we do the same for each tab and group them in the same table
- ...

---

> Well, it does correspond to the tab id, so how about renaming it to that?

It has no relation. It is only there so that the first tab can be marked as `checked`.

---

I still don't like that we're making things too complex than they should (with zipping), especially since we'd only ever need the first id. If you're ok with this, we can simply use jinja:

```nix
''
  <div class=tabs>
    {% set input_group_name = unique_id() %}
    ${lib.concatMapStringsSep "\n" (instruction: ''
      {% set input_id = unique_id() %}
      <input id="{{ input_id }}" type="radio" name="{{ input_group_name }}" {% if input_id == "id-1" %} checked {% endif %}>
      <label for="{{ input_id }}" >${instruction.platform}</label>
      <div class="tab-content">
        ${toString instruction.shell-session}
      </div>
    '') self.instructions}
  </div>
''
```